### PR TITLE
 [ADD] 'hr_timesheet_product_is_employee'

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
# Manage products representing employees

This module was written to extend the Timesheet capabilities of Odoo.

It allows to restrict the access to the products that represent an actual
employee, only to the Human Resources / Manager group.

A company may want to record an actual hourly cost per employee, so that the
time entered by employees through timesheets can result in a more realistic
cost, but at the same time want to conceal this information to general
employees.
# Usage

A user in the group 'Human Resources / Manager' should go to the product form
and set the checkbox 'Is Employee'. This product will then be accessible only

### CODE REVIEW

File: hr_timesheet_product_is_employee/model/hr.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    def _getEmployeeProduct(self, cr, uid, context=None):
+        return super(HrAnalyticTimesheet, self)._getEmployeeProduct(
+            cr, SUPERUSER_ID, context=context)
+
+    _defaults = {
+        'product_uom_id': _getEmployeeUnit,
+        'product_id': _getEmployeeProduct,
+        'general_account_id': _getGeneralAccount,
+    }
```
Comment (pedrobaeza) [inline hr_timesheet_product_is_employee/model/hr.py]: Put a final empty line here.

File: hr_timesheet_product_is_employee/security/product_security.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            <field name="name">Employees cannot view products that represent employees</field>
+            <field name="model_id" ref="product.model_product_template"/>
+            <field name="domain_force">[('is_employee', '=', False)]</field>
+            <field name="groups" eval="[(4,ref('base.group_user'))]"/>
+        </record>
+
+        <record id="property_rule_product_employee_hr_manager" model="ir.rule">
+            <field name="name">HR Managers can view products that represent employees</field>
+            <field name="model_id" ref="product.model_product_template"/>
+            <field name="domain_force">[('is_employee', '=', True)]</field>
```
Comment (pedrobaeza) [inline hr_timesheet_product_is_employee/security/product_security.xml]: You should put `[(1, '=', 1)]`

**Comment by pedrobaeza** [comment]
I would change the module name to `hr_employee_product`. It's significant enough and the product cost is not only related to timesheet, but to analytics and so on.

**Comment by pedrobaeza** [comment]
Some PEP8:

```
./hr_employee_product/model/hr.py:22:1: F401 'fields' imported but unused
./hr_employee_product/model/hr.py:30:80: E501 line too long (127 > 79 characters)
```

**Comment by JordiBForgeFlow** [comment]
Done. Thanks!

**Comment by gurneyalex** [comment]
travis red because of #33

**Comment by charbeljc** [comment]
@jbeficent could you rebase this PR please ? I made it run ok on travis (https://travis-ci.org/charbeljc/hr-timesheet/builds/65077987)

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:

**Comment by JordiBForgeFlow** [comment]
@feketemihai Thanks for merging!

**Pull request merged**